docs/config.rst

```diff
@@ -1,11 +1,606 @@
+
+    Example: When tasks or flows are referenced using the `npsp` namespace, CumulusCI fetches the source from the NPSP GitHub repository.
+
+.. code-block:: yaml
+
+    sources:
+      npsp:
+        github: https://github.com/SalesforceFoundation/NPSP
+
+By default, CumulusCI fetches the most recent release, or the default branch if there are no releases.
+
+.. note::
+    In order for this feature to work, the referenced repository needs to be readable (for example, it's public, or CumulusCI's GitHub service is configured with the token of a user who has read access to it).
+
+It's also possible to fetch a specific ``tag``:
+
+.. code-block:: yaml
+
+    sources:
+      npsp:
+        github: https://github.com/SalesforceFoundation/NPSP
+        tag: rel/3.163
+
+or a specific ``commit`` or ``branch``.
+
+When the repo is listed under ``sources``, it's possible to run a flow from NPSP:
+
+.. code-block:: console
+
+    $ cci flow run npsp:install_prod
+
+Or a task:
+
+.. code-block:: console
+
+    $ cci task run npsp:robot
+
+Or even create a new flow that uses a flow from NPSP:
+
+.. code-block:: yaml
+
+    flows:
+      install_npsp:
+        steps:
+          1:
+            flow: npsp:install_prod
+          2:
+            flow: dev_org
+
+This flow uses NPSP's ``install_prod`` flow to install NPSP as a managed package, and then run this project's own ``dev_org`` flow.
+
+
+
+Troubleshooting Configurations
+------------------------------
+
+Use ``cci task info <name>`` and ``cci flow info <name>`` to see how a given task or flow behaves with the current state of configuration.
+
+    Example: The ``util_sleep`` task has a ``seconds`` option with a default value of 5 seconds.
+
+.. code-block:: console
+
+    $ cci task info util_sleep
+    util_sleep
+
+    Description: Sleeps for N seconds
+
+    Class: cumulusci.tasks.util.Sleep
+
+    Command Syntax
+
+        $ cci task run util_sleep
+
+    Options
+
+        -o seconds SECONDS
+        Required
+        The number of seconds to sleep
+        Default: 5
+
+If you want to change the default value to 30 seconds for all projects, add the desired value in your global ``cumulusci.yml`` file located at ``~/.cumulusci/cumulusci.yml``:
+
+.. code-block:: yaml
+
+    tasks:
+        util_sleep:
+            options:
+                seconds: 30
+
+Now ``cci task info util_sleep`` shows a default of 30 seconds.
+
+.. code-block:: console
+
+    $ cci task info util_sleep
+    util_sleep
+
+    Description: Sleeps for N seconds
+
+    Class: cumulusci.tasks.util.Sleep
+
+    Command Syntax
+
+        $ cci task run util_sleep
+
+    Options
+
+        -o seconds SECONDS
+        Required
+        The number of seconds to sleep
+        Default: 30
+
+Being able to display the active configuration for a given task or flow can help with cross-correlating which configuration scope affects a specific scenario.
```

Comment:
```suggestion
Displaying the active configuration for a given task or flow can help with cross-correlating which configuration scope affects a specific scenario.
```

Reply:
"help with cross-correlating" seems jargony. Would something simpler like "help identify" or "help with identifying" work?

Reply:
@Br4nd0R Thoughts?

Reply:
The point of the example is to show that `cci task info` and `cci flow info` display whatever is currently configured. This example demonstrates that point (showing that if we update a particular `cumulusci.yml` file that those changes are reflected in `cci task info <task_name>`. 

Maybe a tip admonition would be better: 
.. tip::
    The ``cci task info`` and ``cci flow info`` commands will always show information about how a task or flow is *currently* configured. The information output by these commands will change as you make further customizations to your project's ``cumulusci.yml`` file.